react/features/recording/components/RecordingButton.web.js

```diff
@@ -0,0 +1,307 @@
+import AKInlineDialog from '@atlaskit/inline-dialog';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { ToggleStateless } from '@atlaskit/toggle';
+
+import { translate } from '../../base/i18n';
+import { isButtonEnabled, ToolbarButton } from '../../toolbox';
+import Recording from '../../../../modules/UI/recording/Recording';
+
+const DEFAULT_BUTTON_CONFIGURATION = {
+    buttonName: 'recording',
+    classNames: [ 'button', 'icon-recEnable' ],
+    enabled: true,
+    id: 'toolbar_button_record',
+    tooltipKey: 'recording.buttonTooltip'
+};
+
+/**
+ * TOFIX: Copy paste from VideoQualityButton, we need a base class that supports
+ * inline dialogs and does that position thing.
+ * @type {{bottom: string, left: string, right: string, top: string}}
+ */
+const TOOLTIP_TO_DIALOG_POSITION = {
+    bottom: 'bottom center',
+    left: 'left middle',
+    right: 'right middle',
+    top: 'top center'
+};
+
+/**
+ * React {@code Component} for displaying an inline dialog for changing receive
+ * video settings.
+ *
+ * @extends Component
+ */
+class RecordingButton extends Component {
+    /**
+     * {@code RecordingButton}'s property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The redux store representation of the JitsiConference.
+         */
+        _conference: React.PropTypes.object,
+
+        /**
+         * Indicates if the recording button has been toggled.
+         */
+        _recordingButtonToggled: React.PropTypes.bool,
```

Comment:
I'm surprised recording and transcription work differently. Does the transcription service not give a status on success/fail?

Reply:
It only has on/off for now. I was also planning to request some improvements on the jigasi and lib-jitsi-meet side, so that we can add the other states: pending, failed, etc.